### PR TITLE
DataGrid and TreeList: Add React and Vue snippets for onToolbarPreparing

### PR DESCRIPTION
**Pull request by onebalaban**

### CODE REVIEW

File: api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md [hunk 1]
```diff
@@ -99,7 +98,95 @@ The following code shows how you can customize the toolbar using this function.
...
+
+    export default {
+        components: {
+            DxDataGrid
+        },
+        methods: {
+            onToolbarPreparing(e) {
+                var toolbarItems = e.toolbarOptions.items;
+                // Modifies an existing item
+                $.each(toolbarItems, function(_, item) {
```
Comment (RomanTsukanov) [inline api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md]: Hmmm... jQuery.each in Vue?

File: api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md [hunk 2]
```diff
@@ -99,7 +98,95 @@ The following code shows how you can customize the toolbar using this function.
...
+                    }
+                });
+
+                // Adds a new item
+                toolbarItems.push({
+                    widget: 'dxButton',
+                    options: {
+                        icon: 'user',
+                        onClick: function() {
+                            ...
```
Comment (RomanTsukanov) [inline api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md]:
```suggestion
                           //  ...
```

File: api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md [hunk 3]
```diff
@@ -99,7 +98,95 @@ The following code shows how you can customize the toolbar using this function.
...
+
+    import { DxDataGrid } from 'devextreme-vue/data-grid';
+
+    export default {
+        components: {
+            DxDataGrid
+        },
+        methods: {
+            onToolbarPreparing(e) {
+                var toolbarItems = e.toolbarOptions.items;
```
Comment (RomanTsukanov) [inline api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md]:
```suggestion
                let toolbarItems = e.toolbarOptions.items;
```

File: api-reference/10 UI Widgets/dxTreeList/1 Configuration/onToolbarPreparing.md [hunk 4]
```diff
@@ -20,11 +20,11 @@ The following code shows how you can customize the toolbar using this function.
             onToolbarPreparing: function (e) {
                 var toolbarItems = e.toolbarOptions.items;
                 // Modifies an existing item
-                toolbarItems.forEach(function(item) {
-                    if (item.name === "saveButton") {
+                $.each(toolbarItems, function(_, item) {
```
Comment (RomanTsukanov) [inline api-reference/10 UI Widgets/dxTreeList/1 Configuration/onToolbarPreparing.md]: `forEach` is part of JavaScript. Why use `jQuery.each`?

File: api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md [hunk 5]
```diff
@@ -66,7 +66,7 @@ The following code shows how you can customize the toolbar using this function.
 ##### Angular
 
     <!--TypeScript-->
-    import { DxDataGridModule, DxButtonModule } from "devextreme-angular";
+    import { DxDataGridModule } from "devextreme-angular";
```
Comment (RomanTsukanov) [inline api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md]: Are you sure that the Button widget will be imported automatically once I add a button toolbar item?

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md (hunk 1 above)
Comment (onebalaban) [reply api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md]: Tested our code converter. It's just a bug.

File: api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md [hunk 6]
```diff
@@ -99,7 +99,96 @@ The following code shows how you can customize the toolbar using this function.
...
+        <DxDataGrid ...
+            @toolbar-preparing="onToolbarPreparing"
+        />
+    </template>
+
+    <script>
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { DxDataGrid, DxButton } from 'devextreme-vue/data-grid';
```
Comment (RomanTsukanov) [inline api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md]: Are you sure that named import is needed in this case? I thought it should be something like this:

```suggestion
    import { DxDataGrid } from 'devextreme-vue/data-grid';
    import 'devextreme-vue/button';
```

File: api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md [hunk 7]
```diff
@@ -99,7 +99,96 @@ The following code shows how you can customize the toolbar using this function.
...
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { DataGrid, Button } from 'devextreme-react/data-grid';
```
Comment (RomanTsukanov) [inline api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md]: Same note here.

File: api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md [hunk 8]
```diff
@@ -99,7 +99,96 @@ The following code shows how you can customize the toolbar using this function.
...
+    <script>
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { DxDataGrid, DxButton } from 'devextreme-vue/data-grid';
+
+    export default {
+        components: {
+            DxDataGrid,
+            DxButton
```
Comment (RomanTsukanov) [inline api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md]: `DxButton` is not used in the markup.

```suggestion
            DxDataGrid
```

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md (hunk 6 above)
Comment (onebalaban) [reply api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md]: Recently, in a similar case in the DataGrid popup editing demo, we faced the issue that DxButton should be imported in React and Vue. So, I added it to code snippets.

File: api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md (hunk 7 above)
Comment (onebalaban) [reply api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md]: Recently, in a similar case in the DataGrid popup editing demo, we faced the issue that DxButton should be imported in React and Vue. So, I added it to code snippets.

File: api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md (hunk 8 above)
Comment (onebalaban) [reply api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md]: Same note here.

File: api-reference/10 UI Widgets/dxTreeList/1 Configuration/onToolbarPreparing.md [hunk 9]
```diff
@@ -72,5 +72,93 @@ The following code shows how you can customize the toolbar using this function.
...
+        <DxTreeList ...
+            @toolbar-preparing="onToolbarPreparing"
+        />
+    </template>
+
+    <script>
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { DxDataGrid } from 'devextreme-vue/tree-list';
```
Comment (RomanTsukanov) [inline api-reference/10 UI Widgets/dxTreeList/1 Configuration/onToolbarPreparing.md]:
```suggestion
    import { DxTreeList } from 'devextreme-vue/tree-list';
```

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

**Review by RomanTsukanov — APPROVED** [review]

**Review by artem-kurchenko — APPROVED** [review]

**Pull request merged**